src/utilities/apiImpl.ts

```diff
@@ -0,0 +1,52 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ */
+
+import * as sarifApi from "./../api/sarifViewerApi";
+import * as vscode from "vscode";
+import { LogReader } from "../logReader";
+import { SVDiagnosticCollection } from "../svDiagnosticCollection";
+import { openSarifFile } from "../extension";
+
+export class ApiImpl implements sarifApi.Api {
+    /**
+     * Constructs an instance of the API given out by the extension.
+     * @param logReader An instance of the log reader to use for parsing log files.
+     * @param diagnosticCollection The diagnostic collection that holds the results.
+     */
+    public constructor(private readonly logReader: LogReader, private readonly diagnosticCollection: SVDiagnosticCollection) {
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public async openLogs(sarifFiles: vscode.Uri[], openLogFileArguments?: sarifApi.OpenLogArguments): Promise<void> {
```

Comment:
>Logs [](start = 21, length = 4)

nit: would it be worth specifying what type of logs (e.g. `openSarifFiles`)? or is it obvious that these are sarif files?

Reply:
Nope. you are already using the SARIF apis, so that would be redundant

---
In reply to: [409154113](https://github.com/microsoft/sarif-vscode-extension/pull/248#discussion_r409154113) [](ancestors = 409154113)

Reply:
Another nit: param `sarifFiles` should technically be `uris` or `logUris`.

"SARIF file" is a colloquialism.

Reply:
@sherryyshi There is only one type.